### ISSUE TITLE
/i freezes when holding a container with more items than it can hold

### ISSUE THREAD

**Issue by Zbizu** [otland/forgottenserver#3225]
tl;dr: a bag with 10/8 items inside will freeze the server for a few seconds when you try to /i something that can't be picked up (eg. a wall or a ground)

- [x] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [x] This issue is reproducible without changes to the C++ code in this repository

### Steps to reproduce
1. go to items.xml, set a bag volume to 10
2. have this bag in your eq
3. put 10 items in it
4. save server and shut it down
5. set bag volume back to 8
6. start your server again
7. try to /i a ground tile or a wall
8. wait a few seconds to log in again once it unfreezes
9. throw 2 items out of this bag
10. /i works normally

### Expected behaviour
created wall/ground is on the floor without freezing

### Actual behaviour
server freezes

### Environment
windows 7
compilation mode release 64-bit
tfs sources from Nov 10 2020 (freepremium fix commit)
visual studio 2017, using vcpkg setup
v141, sdk version 10.0.18362.0
boost version 1_71 x64

This bug happened in both - tfs compiled by myself and appveyor build.

**Comment by Zbizu** [comment]
also leaving a reference to #3222 for clarity

**Comment by kygov** [comment]
I'm not sure, it may be because of uint32_t overflow here.
https://github.com/otland/forgottenserver/blob/master/src/player.cpp#L2644

**Comment by EPuncker** [comment]
could you test these https://github.com/SaiyansKing/optimized_forgottenserver/commit/a205b5f033e88fe217721116dd623b6ef253bb43

and also open a PR if its the case
